apps/keyboard/test/marionette/lib/keyboard.js

```diff
@@ -97,28 +127,25 @@ Keyboard.prototype = {
     return 2;
   },
 
-  switchCase: function switchCase(upperCase) {
-    if (this.isUpperCase() === upperCase) {
-      return;
-    }
+  getSuggestionKey: function(word) {
+    var selector = Keyboard.Selector.suggestionKey.replace(/%s/, word);
+    return this.client.findElement(selector);
+  },
 
-    var shiftKey = this.shiftKey;
+  getCurrentKeyboard: function() {
+    var activeFrame = this.activeKeyboardFrame;
 
-    shiftKey.tap();
-    this.client.waitFor(function() {
-      var expected = upperCase ? 'true' : 'false';
-      return (shiftKey.getAttribute('aria-pressed') === expected);
-    });
+    return activeFrame.getAttribute('data-frame-name');
   },
 
-  switchToPage: function(index) {
-    var pageIndex = this.getCurrentPageIndex();
-    if (pageIndex === index) {
-      return;
-    }
+  getCurrentInputType: function () {
+    return this.client.executeScript(
+      'return window.wrappedJSObject.app.getBasicInputType();');
+  },
 
-    var pageSwitchingKey = this.getPageSwitchingKey(index);
-    pageSwitchingKey.tap();
+  getCurrentInputMode: function () {
+    return this.client.executeScript(
+      'return window.wrappedJSObject.app.inputContext.inputMode;');
```

Comment:
`wrappedJSObject`: Probing JS values in the app context can only be used as helper function, not something the tests should assert against.

I am fine if we reimpl these functions from Gip but please make sure assertions are added accordingly (like my previous comment).